fvtest/jitbuildertest/GlobalTest.cpp

```diff
@@ -0,0 +1,149 @@
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#include "JBTestUtil.hpp"
+
+static int64_t globalInt64 = 0;
+
+static int32_t globalInt32 = 0;
+
+static int16_t globalInt16 = 0;
+
+static int8_t globalInt8 = 0;
+
+DEFINE_BUILDER(TestInt64Global,
+               Int64,
+               PARAM("param", Int64))
+   {
+   DefineGlobal("int64global",Int64,(void*)(&globalInt64));
+
+   OMR::JitBuilder::IlValue *param = Load("param");
+   Store("int64global",param);
+   OMR::JitBuilder::IlValue *int64global = Load("int64global");
+   OMR::JitBuilder::IlValue *int64Val = ConstInt64(1);
+   OMR::JitBuilder::IlValue *result = Add(int64global, int64Val);
+
+   Return(result);
+
+   return true;
+   }
+
+DEFINE_BUILDER(TestInt32Global,
+               Int32,
+               PARAM("param", Int32))
+   {
+
+   DefineGlobal("int32global",Int32,(void*)(&globalInt32));
+
+   OMR::JitBuilder::IlValue *param = Load("param");
+   Store("int32global",param);
+   OMR::JitBuilder::IlValue *int32global = Load("int32global");
+   OMR::JitBuilder::IlValue *int32Val = ConstInt32(1);
+   OMR::JitBuilder::IlValue *result = Add(int32global, int32Val);
+
+   Return(result);
+
+   return true;
+   }
+
+DEFINE_BUILDER(TestInt16Global,
+               Int16,
+               PARAM("param", Int16))
+   {
+
+   DefineGlobal("int16global",Int16,(void*)(&globalInt16));
+
+   OMR::JitBuilder::IlValue *param = Load("param");
+   Store("int16global",param);
+   OMR::JitBuilder::IlValue *int16global = Load("int16global");
+   OMR::JitBuilder::IlValue *int16Val = ConstInt16(1);
+   OMR::JitBuilder::IlValue *result = Add(int16global, int16Val);
+
+   Return(result);
+
+   return true;
+   }
+
+DEFINE_BUILDER(TestInt8Global,
+               Int8,
+               PARAM("param", Int8))
+   {
+
+   DefineGlobal("int8global",Int8,(void*)(&globalInt8));
+
+   OMR::JitBuilder::IlValue *param = Load("param");
+   Store("int8global",param);
+   OMR::JitBuilder::IlValue *int8global = Load("int8global");
+   OMR::JitBuilder::IlValue *int8Val = ConstInt8(1);
+   OMR::JitBuilder::IlValue *result = Add(int8global, int8Val);
+
+   Return(result);
+
+   return true;
+
+   }
```

Comment:
I'd like to at least see a pointer type being used, and completely would suggest having Float and Double types too...

Reply:
Address, Float and Double tests added in https://github.com/eclipse/omr/pull/4774/commits/da04cd989c8b031d3444489d397f6297da254535.